src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -134,5 +127,268 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
+                    data,
+                    parts,
+                    unconvertedInterpolatedString.ConstantValue,
+                    unconvertedInterpolatedString.Type,
+                    unconvertedInterpolatedString.HasErrors);
+
+            bool tryBindAsBuilderType([NotNullWhen(true)] out BoundInterpolatedString? result)
+            {
+                if (this.Flags.Includes(BinderFlags.InCatchFilter))
+                {
+                    // CLI Spec does not permit nested tries inside a filter block, so we can't use the builder here, as it is disposable.
+                    // PROTOTYPE(interp-string): Should we try to collect errors if a type not otherwise usable in an interpolated string
+                    // is used in this case?
+                    result = null;
+                    return false;
+                }
+
+                result = null;
+                var interpolatedStringBuilderType = Compilation.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_InterpolatedStringBuilder);
+                if (interpolatedStringBuilderType is MissingMetadataTypeSymbol)
+                {
+                    return false;
+                }
+
+                if (unconvertedInterpolatedString.Parts.ContainsAwaitExpression())
+                {
+                    return false;
+                }
+
+                BindingDiagnosticBag applicableDiagnostics = BindingDiagnosticBag.GetInstance(template: diagnostics);
+                if (!Conversions.IsApplicableInterpolatedStringBuilderType(unconvertedInterpolatedString, interpolatedStringBuilderType, applicableDiagnostics, out var builderArguments))
+                {
+                    applicableDiagnostics.Free();
+                    return false;
+                }
+
+                diagnostics.AddRangeAndFree(applicableDiagnostics);
+
+                // Prior to C# 10, all types in an interpolated string expression needed to be convertible to `object`. After 10, some types
+                // (such as Span<T>) that are not convertible to `object` are permissible as interpolated string components, provided there
+                // is an applicable TryFormatInterpolationHole that accepts them. To preserve langversion, we therefore make sure all components
+                // are convertible to object if the current langversion is lower than the interpolation feature
+                TypeSymbol? objectType = null;
+                BindingDiagnosticBag? conversionDiagnostics = null;
+                var needToCheckConversionToObject = !Compilation.IsFeatureEnabled(MessageID.IDS_FeatureImprovedInterpolatedStrings) && diagnostics.AccumulatesDiagnostics;
+                if (needToCheckConversionToObject)
+                {
+                    objectType = GetSpecialType(SpecialType.System_Object, diagnostics, unconvertedInterpolatedString.Syntax);
+                    conversionDiagnostics = BindingDiagnosticBag.GetInstance();
```

Comment:
>GetInstance [](start = 65, length = 11)

Is it necessary to track dependencies?

Reply:
I meant to use the input as a template, thanks.

---
In reply to: [610909474](https://github.com/dotnet/roslyn/pull/52422#discussion_r610909474) [](ancestors = 610909474)